pkg/controller/master/setting/kubevirt_migration.go

```diff
@@ -0,0 +1,54 @@
+package setting
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	kubevirtv1 "kubevirt.io/api/core/v1"
+
+	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester/pkg/util"
+)
+
+func (h *Handler) syncKubeVirtMigration(setting *harvesterv1.Setting) error {
+	var value string
+	if setting.Value != "" {
+		value = setting.Value
+	} else {
+		value = setting.Default
+	}
+
+	var err error
+	migrationConfiguration := &kubevirtv1.MigrationConfiguration{}
+
+	if value != "" {
+		err = json.Unmarshal([]byte(value), migrationConfiguration)
+		if err != nil {
+			return fmt.Errorf("invalid value: `%s`: %w", value, err)
+		}
+	}
+
+	kubevirt, err := h.kubeVirtConfigCache.Get(util.HarvesterSystemNamespaceName, util.KubeVirtObjectName)
+	if err != nil {
+		return fmt.Errorf("failed to get kubevirt object %v/%v %w", util.HarvesterSystemNamespaceName, util.KubeVirtObjectName, err)
+	}
+
+	kubevirtCpy := kubevirt.DeepCopy()
+	if kubevirtCpy.Spec.Configuration.MigrationConfiguration == nil {
+		kubevirtCpy.Spec.Configuration.MigrationConfiguration = &kubevirtv1.MigrationConfiguration{}
+	}
+	// ignore nodeDrainTaintKey and network field
+	// The default nodeDrainTaintKey is "kubevirt.io/drain" and it's used in upgrade script.
+	// The network field is handled by "vm-migration-network" setting.
+	migrationConfiguration.NodeDrainTaintKey = nil
+	migrationConfiguration.Network = kubevirtCpy.Spec.Configuration.MigrationConfiguration.Network
```

Comment:
Is it possible that users might have their own setting?

Reply:
We already have vm-migration-network for this setting. It's better to avoid unexpected value here and just used value from vm-migration-network setting.

https://docs.harvesterhci.io/v1.6/advanced/vm-migration-network